### PR TITLE
clean up code

### PR DESCRIPTION
**Pull request by wiederm**
## Description
This PR updates some of the legacy code that has been part of this repository as a result of some ad hoc decisions during the begging of the project.

Specifically, (1) we are now using the dedicated DCDReporter for saving trajectories, (2) handle the system generation outside of the package and only generate the simulation instance, (3) automatically detect the fastest platform, (4) remove functions for velocity seeding (now use the openMM functionality for that)

## Status
- [x] Ready to go

### CODE REVIEW

File: endstate_correction/protocol.py [hunk 1]
```diff
@@ -107,28 +110,27 @@ class Protocol:
 
     method: str
     sim: Simulation
-    trajectories: List
-    direction: str = "None"
+    reference_samples: md.Trajectory
+    target_samples: Union[md.Trajectory, None] = None
```
Comment (xiki-tempula) [inline endstate_correction/protocol.py:114]: `Optional[md.Trajectory]`

Comment (wiederm) [reply endstate_correction/protocol.py:114]: that's great, thanks! will change!

File: endstate_correction/system.py [hunk 2]
```diff
@@ -74,61 +61,58 @@ def read_box(psf: CharmmPsfFile, filename: str) -> CharmmPsfFile:
...
-    ml_system = potential.createMixedSystem(
-        psf.topology, mm_system, ml_atoms, interpolate=True
-    )
-    #####################
-
-    integrator = mm.LangevinIntegrator(temperature, collision_rate, stepsize)
-    platform = mm.Platform.getPlatformByName(platform)
-
-    return Simulation(psf.topology, ml_system, integrator, platform=platform)
+# def create_charmm_system(
```
Comment (xiki-tempula) [inline endstate_correction/system.py:64]: Why not just remove this?

**Comment by wiederm** [comment]
Thanks, @xiki-tempula for your input. I agree that this PR should have been split into multiple, smaller PRs. The process here was a bit rushed and doesn't complelty reflect best practices.

**Pull request merged**